### PR TITLE
Rewritten examples with ES6

### PR DESCRIPTION
**Pull request by SuperPaintman**
Hi!
I rewrote the examples using the ES6, and format them.

### CODE REVIEW

**Comment by oleavr** [comment]
Awesome, thanks for doing this! Would you also be interested in converting the last details also? Off the top of my head:

- Get rid of `var` declarations and replace them by either `const` or `let`
- Convert `{ onEnter: function (args) {}, onLeave: function (retval) {} }` to `{ onEnter(args) {}, onLeave(retval) {} }`

Cheers!

**Comment by SomeoneWeird** [comment]
Also, change the scripts and use backticks instead of doing `"" + "" +` etc :tada:

**Comment by SuperPaintman** [comment]
@oleavr, @SomeoneWeird Done :)

**Comment by oleavr** [comment]
Sweet! It would be great to also omit the space between `onEnter` and `(`, and perhaps also remove the vertical alignment (it was there already, but I just thought of it now), so the code layout is stable even if a variable is renamed later.

**Comment by SuperPaintman** [comment]
I removed the space

**Comment by oleavr** [comment]
Nice work! What do you think about getting rid of the last `var` occurences also?

**Comment by SuperPaintman** [comment]
@oleavr they can't be replaced by a `const`, because we replace placeholder '%addr%'.

**Comment by oleavr** [comment]
@SuperPaintman That's fine, `const` just means it's not possible to assign a new value to the variable, so in such cases we should use `let` (`var` is legacy). The '%addr%' placeholder is replaced when the script is generated, but before it is loaded, so even those variables should be `const` (as long as we never reassign, or need to declare the variable without assigning in order to assign it later).

**Comment by SuperPaintman** [comment]
@oleavr Oh, sorry, I do not fully understood the concept of the new variables in ES6. Now i replaced `var` to `let`

File: examples/blocking_receives.js [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
-"  }"                                           +
-"});";
+const processName    = process.argv[2];
+const processAddress = process.argv[3];
 
-frida.attach(processName).then(function (session) {
-  return session.createScript(script.replace("%addr%", processAddress));
-}).then(function (script) {
-  script.events.listen('message', function (message) {
+let script =
```
Comment (oleavr) [inline examples/blocking_receives.js]: This should be `const` (it's never reassigned a new value).

File: examples/blocking_receives.js [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
+  }
+});`;
+
+frida.attach(processName)
+.then(session => session.createScript(script.replace("%addr%", processAddress)))
+.then(script => {
+  script.events.listen('message', message => {
     console.log(message);
-    var val = parseInt(message.payload);
+    let val = parseInt(message.payload);
```
Comment (oleavr) [inline examples/blocking_receives.js]: This should be const (it's never reassigned a new value).

File: examples/calling_functions.js [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-frida.attach(processName).then(function (session) {
-  return session.createScript(script.replace('%addr%', processAddress));
-}).then(function (script) {
-  script.load().then(function () {
-    console.log("script loaded");
-  }).catch(function (err) {
-    console.error(err);
-  });
-}).catch(function (err) {
+let script =
```
Comment (oleavr) [inline examples/calling_functions.js]: This should be const (it's never reassigned a new value).

File: examples/hook_function.js [hunk 4]
```diff
@@ -1,24 +1,29 @@
...
-"  }"                                 +
-"});";
+const processName    = process.argv[2];
+const processAddress = process.argv[3];
 
-frida.attach(processName).then(function (session) {
-  return session.createScript(script.replace('%addr%', processAddress));
-}).then(function (script) {
-  script.events.listen('message', function (message) {
+let script =
```
Comment (oleavr) [inline examples/hook_function.js]: This should be const (it's never reassigned a new value).

File: examples/interactive.js [hunk 5]
```diff
@@ -1,51 +1,60 @@
-var frida = require('..');
+'use strict';
+
+const frida = require('..');
+
+let scriptText =
```
Comment (oleavr) [inline examples/interactive.js]: This should be const (it's never reassigned a new value).

File: examples/modify_arguments.js [hunk 6]
```diff
@@ -1,24 +1,29 @@
...
-"  }"                                 +
-"});";
+const processName    = process.argv[2];
+const processAddress = process.argv[3];
 
-frida.attach(processName).then(function (session) {
-  return session.createScript(script.replace('%addr%', processAddress));
-}).then(function (script) {
-  script.events.listen('message', function (message) {
+let script =
```
Comment (oleavr) [inline examples/modify_arguments.js]: This should be const (it's never reassigned a new value).

File: examples/ping_pong.js [hunk 7]
```diff
@@ -1,24 +1,27 @@
...
-"recv('poke', function onMessage(pokeMessage) {" +
-"  send('pokeBack');"                            +
-"});";
+const processName = process.argv[2];
 
-frida.attach(processName).then(function (session) {
-  return session.createScript(script);
-}).then(function (script) {
-  script.events.listen('message', function (message) {
+let script =
```
Comment (oleavr) [inline examples/ping_pong.js]: This should be const (it's never reassigned a new value).

File: examples/sending_messages.js [hunk 8]
```diff
@@ -1,20 +1,23 @@
...
+const frida = require('..');
 
-var script = "send(1337);";
+const processName = process.argv[2];
 
-frida.attach(processName).then(function (session) {
-  return session.createScript(script);
-}).then(function (script) {
-  script.events.listen('message', function (message) {
+let script = "send(1337);";
```
Comment (oleavr) [inline examples/sending_messages.js]: This should be const (it's never reassigned a new value).

File: examples/interactive.js [hunk 9]
```diff
@@ -1,51 +1,60 @@
...
-        script.postMessage({ name: 'ping' });
-      }, 1000);
-    });
+
+    return script.load();
+  })
+  .then(() => {
+    console.log('script loaded');
+    setInterval(() => {
+      script.postMessage({ name: 'ping' });
```
Comment (oleavr) [inline examples/interactive.js]: This looks like a regression – the `script` variable does not exist here.

File: examples/ping_pong.js [hunk 10]
```diff
@@ -1,24 +1,27 @@
...
-  }).catch(function (err) {
-    console.error(err);
-  });
-}).catch(function (err) {
+
+  return script.load();
+})
+.then(() => {
+  console.log("script loaded");
+  script.postMessage({ "type": "poke" });
```
Comment (oleavr) [inline examples/ping_pong.js]: This looks like a regression – the `script` variable is bound to the source-code and not the script that was returned by `session.createScript()`.

**Comment by oleavr** [comment]
@SuperPaintman No worries! I've added some more comments on the issues I noticed. By the way, maybe we also want to switch the examples to using `co`? That way we'd get rid of the callback chaining spaghetti that we have had so far (because we couldn't use `co` due to lack of ES6 support). Here's an example: https://gist.github.com/oleavr/a22d675b76e7509cd2c9#file-app-js

**Comment by oleavr** [comment]
And I also noticed that maybe we want to rename `script` to `source`, to distinguish the source-code from the resulting script object (it's also named `source` at the API level).

**Comment by SuperPaintman** [comment]
@oleavr thanks for suggested to learn the CO. This is probably my best experience this week!

**Pull request merged**